### ISSUE TITLE
Game request: [calme][160628]KARAKARA

### ISSUE THREAD

**Issue by WorkingWarrior** [vn-tools/arc_unpacker#47]
It seems that XP3 archives use different encryption key.
Could you, please, add support for it?
https://vndb.org/v18974

**Comment by rr-** [comment]
The game uses CXDEC and it looks different this time. It seems to process the whole file in one go.
Might be trickier to implement it.

**Comment by rr-** [comment]
This is the old CXDEC in a new skin. I couldn't find the decryption parameters (namely the keys used to derive the "edge" between two encryption passes) in the game code, but I managed to obtain them via brute force nonetheless. This is surprising because these keys were easiest to get in previous CXDEC implementations.

@morkt it also seems as if CXDEC control block discovery needs adjustments:

1. I couldn't find physical `cxdec.tpm` (I haven't bothered to look how the game actually ends up loading it).
2. The block itself no longer begins with ` Encryption control block` and contains random garbage instead.
3. It's immediately followed by `insufficient code space` error message. Maybe this could be used to search for it.

Eventually rather than writing extra discovery bloat, I ended up just embedding the control block into the program.


Changes should be available in tonight's nightly build:

- 0e63d812196b56bdac760efe4c06295bd86c772f
- ab98ab2aa1be2bc7ad050217b14cd63d84b220a9
- 94f780c048dc128fe8999afef5fe9de04becaac2
